dcpy/connectors/edm/packaging.py

```diff
@@ -0,0 +1,150 @@
+import os
+import shutil
+from pathlib import Path
+from typing import Callable
+from dataclasses import dataclass
+import typer
+
+from dcpy.utils import s3
+from dcpy.utils.logging import logger
+from dcpy.connectors.edm import publishing
+
+BUCKET = "edm-distributions"
+BUCKET_ACL = s3.string_as_acl("public-read")
+
+DOWNLOAD_ROOT_PATH = Path(".library")
+OUTPUT_ROOT_PATH = Path(".output")
+
+
+@dataclass
+class PackageKey(publishing.ProductKey):
+    product: str
+    version: str
+
+    def __str__(self):
+        return f"{self.product} - {self.version}"
+
+    @property
+    def path(self) -> str:
+        return f"{self.product}/{self.version}"
+
+
+def colp(publish_key: publishing.PublishKey, package_key: PackageKey):
+    raise NotImplementedError
+
+
+def templatedb(publish_key: publishing.PublishKey, package_key: PackageKey):
+    output_path = OUTPUT_ROOT_PATH / package_key.path
+    shutil.copytree(
+        DOWNLOAD_ROOT_PATH / publish_key.path,
+        output_path,
+        dirs_exist_ok=True,
+    )
+    shutil.copy(
+        output_path / "templatedb.csv",
+        output_path / "templatedb_packaged.csv",
+    )
+
+
+@dataclass
+class PackageMetadata:
+    product: str
+    packaged_name: str
+    packaging_function: Callable
+
+
+DATASET_PACKAGE_METADATA = {
+    "db-colp": PackageMetadata("db-colp", "dcp_colp", colp),
+    "db-template": PackageMetadata("db-colp", "dcp_template_db", templatedb),
```

Comment:
Wrong product!

Reply:
good thing nothing uses `PackageMetadata.product! maybe a sign to drop the field until it's useful